File: app/src/main/java/edu/sunypoly/inventoryapp/ItemRecyclerAdapter.java
package edu.sunypoly.inventoryapp;

import android.support.annotation.NonNull;
import android.support.constraint.ConstraintLayout;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.ViewGroup;

import java.util.ArrayList;

public class ItemRecyclerAdapter extends RecyclerView.Adapter<ItemViewHolder> {

    private ArrayList<InventoryItem> mDataset;

    public ItemRecyclerAdapter(ArrayList<InventoryItem> dataset){
        mDataset = dataset;
    }

    public ArrayList<InventoryItem> getmDataset() {
        return mDataset;
    }

    public void setmDataset(ArrayList<InventoryItem> mDataset) {
        this.mDataset = mDataset;
    }

    @NonNull
    @Override
    public ItemViewHolder onCreateViewHolder(@NonNull ViewGroup viewGroup, int i) {
        ConstraintLayout v = (ConstraintLayout) LayoutInflater.from(viewGroup.getContext())
                .inflate(R.layout.editable_list_item, viewGroup, false);

        return new ItemViewHolder(v);
    }

    @Override
    public void onBindViewHolder(@NonNull ItemViewHolder itemViewHolder, int i) {
        InventoryItem item = mDataset.get(i);
        String line = "Name: " + item.getName() + "\n\tRoom: " + item.getRoom() +
                "\n\tAsset tag: " + item.getBarcode();
        itemViewHolder.getItemTextView().setText(line);
        itemViewHolder.setInventoryItem(mDataset.get(i));
    }

    @Override
    public int getItemCount() {
        return mDataset.size();
    }
}

File: app/src/main/java/edu/sunypoly/inventoryapp/ItemRecyclerAdapter.kt
package edu.sunypoly.inventoryapp

import android.os.Handler
import androidx.constraintlayout.widget.ConstraintLayout
import androidx.recyclerview.widget.RecyclerView
import android.view.LayoutInflater
import android.view.ViewGroup
import java.lang.reflect.InvocationHandler

import java.util.ArrayList

/**
 * Handles inflation of inventoryItems; connects ItemViewHolder to ListAllItemsActivity or SearchActivity
 */
class ItemRecyclerAdapter(val handler: Handler, var mDataset: ArrayList<InventoryItem>?) : androidx.recyclerview.widget.RecyclerView.Adapter<ItemViewHolder>() {

    fun getmDataset(): ArrayList<InventoryItem>? {
        return mDataset
    }

    fun setmDataset(mDataset: ArrayList<InventoryItem>) {
        this.mDataset = mDataset
    }

    override fun onCreateViewHolder(viewGroup: ViewGroup, i: Int): ItemViewHolder {
        val v = LayoutInflater.from(viewGroup.context)
                .inflate(R.layout.editable_list_item, viewGroup, false) as ConstraintLayout

        return ItemViewHolder(handler, v)
    }

    override fun onBindViewHolder(itemViewHolder: ItemViewHolder, i: Int) {
        val item = mDataset!![i]
        val line = "Name: " + item.name + "\n\tRoom: " + item.room +
                "\n\tAsset tag: " + item.barcode
        itemViewHolder.itemTextView.text = line
        itemViewHolder.inventoryItem = mDataset!![i]
    }

    override fun getItemCount(): Int {
        return mDataset!!.size
    }
}
